libs/application/core/src/validation/validators.ts

```diff
@@ -49,37 +53,45 @@ function partialSchemaValidation(
     try {
       trimmedSchema.parse({ [key]: answer })
     } catch (e) {
-      error = populateError(error, e, newPath)
-
+      const zodErrors: ZodSuberror[] = e.errors
+      const keyIsIncludedInErrors = zodErrors.some((err) =>
+        err.path.includes(key),
+      )
+      if (!has(error, constructedErrorPath) && keyIsIncludedInErrors) {
+        error = populateError(
+          error,
+          zodErrors,
+          sendConstructedPath ? constructedErrorPath : undefined,
+        )
+      }
       if (Array.isArray(answer)) {
         const arrayElements = answer as Answer[]
         arrayElements.forEach((el, index) => {
           try {
             trimmedSchema.parse({ [key]: [el] })
           } catch (e) {
-            const elementPath = `${newPath}[${index}]`
-            error = populateError(error, e, elementPath)
+            const elementPath = `${constructedErrorPath}[${index}]`
```

Comment:
Is `indexedPath` a better name here? Since that is the only thing we are adding to the element path?